client/rest/test/plugins/rosetta/nem/blockRoutes_spec.js

```diff
@@ -0,0 +1,278 @@
+import { RosettaErrorFactory } from '../../../../src/plugins/rosetta/rosettaUtils.js';
+import { RosettaOperationFactory } from '../utils/rosettaTestUtils.js';
+import { NemFacade } from 'symbol-sdk/nem';
+
+describe('NEM block routes', () => {
+	const assertRosettaErrorRaisedBasic = (...args) => assertRosettaErrorRaisedBasicWithRoutes(blockRoutes, ...args);
+	const assertRosettaSuccessBasic = (...args) => assertRosettaSuccessBasicWithRoutes(blockRoutes, ...args);
+
+	// region utils
+
+	const TEST_BLOCK_HEIGHT = 1111;
+	const TRANSACTION_HASH = 'A5DEA9FA1CB0C88EB9EC9156602C68902FB2F61E13342E552603DC55FF199D89';
+	const RECIPIENT_ADDRESS = 'TCYGRS7EWWHSNFUIETQIMB233NE75NVX6MJFI3JO';
+	const SIGNER_PUBLICKEY = '61127CD45073DFED58F472748900D0B90E2D8EC6E10EE4E41B03861D1D1A720D';
+	const SIGNER_ADDRESS = 'TCATGBEBY4GVVDEBDJCCSCYNK3XWORFB3R4PPZZU';
+
+	const { createRosettaNetworkIdentifier } = RosettaObjectFactory;
+	const createTransferOperation = (...args) =>
+		RosettaOperationFactory.setOperationStatus(RosettaOperationFactory.createTransferOperation(...args));
+
+	// endregion
+
+	// region stub helpers
+
+	const stubFetchResult = FetchStubHelper.stubPost;
+	FetchStubHelper.registerStubCleanup();
+
+	const createTransactionJson = (height = TEST_BLOCK_HEIGHT) => {
+		const facade = new NemFacade('testnet');
+		const textEncoder = new TextEncoder();
+		const transaction = facade.transactionFactory.create({
+			type: 'transfer_transaction_v2',
+			recipientAddress: RECIPIENT_ADDRESS,
+			signerPublicKey: SIGNER_PUBLICKEY,
+			amount: 4_000000,
+			fee: 1000,
+			mosaics: [
+				{
+					mosaic: {
+						mosaicId: { namespaceId: { name: textEncoder.encode('nem') }, name: textEncoder.encode('xem') },
+						amount: 10_000000n
+					}
+				},
+				{
+					mosaic: {
+						mosaicId: {
+							namespaceId: { name: textEncoder.encode('magic') },
+							name: textEncoder.encode('hat')
+						},
+						amount: 5_00n
+					}
+				}
+			]
+		});
+
+		return {
+			transaction: convertTransactionSdkJsonToRestJson(transaction.toJson()),
+			meta: {
+				hash: { data: TRANSACTION_HASH },
+				height
+			}
+		};
+	};
+
+	const createMatchingRosettaTransaction = () => {
+		const transferCurrencyProperties = ['magic.hat', 3];
+		const feeCurrencyProperties = ['nem.xem', 6];
+		return new Transaction(
+			new TransactionIdentifier(TRANSACTION_HASH),
+			[
+				createTransferOperation(0, SIGNER_ADDRESS, '-40000000', ...feeCurrencyProperties),
+				createTransferOperation(1, RECIPIENT_ADDRESS, '40000000', ...feeCurrencyProperties),
+				createTransferOperation(2, SIGNER_ADDRESS, '-2000', ...transferCurrencyProperties),
+				createTransferOperation(3, RECIPIENT_ADDRESS, '2000', ...transferCurrencyProperties),
+				createTransferOperation(4, SIGNER_ADDRESS, '-1000', ...feeCurrencyProperties)
+			]
+		);
+	};
+
+	const stubFetchNamespaceResult = (name, id, divisibility) => {
+		stubFetchResult(`namespace/definition/page?namespace=${id}&pageSize=100`, true, {
```

Comment:
stubMosaicResolution

Reply:
we can replace with test helper from other PR depending which is merged first 😅

Reply:
Will wait until you checkin the other PRs

Reply:
actually, i have a few small cleanup, so i'll do it after this is merged